app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,503 @@
+
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+type Props = {
+
+    // Screen Props (do not change during the lifetime of the screen)
+    componentId: string;
+    categoryId?: string;
+    closeButton: ImageResource;
+
+    // Calculated Props
+    canCreateChannels: boolean;
+    currentUserId: string;
+    currentTeamId: string;
+    joinedChannels?: MyChannelModel[];
+    sharedChannelsEnabled: boolean;
+    canShowArchivedChannels: boolean;
+}
+
+const CLOSE_BUTTON_ID = 'close-browse-channels';
+const CREATE_BUTTON_ID = 'create-pub-channel';
+
+const filterChannelsByTerm = (channels: Channel[], term: string) => {
+    const lowerCasedTerm = term.toLowerCase();
+    return channels.filter((c) => {
+        return (c.name.toLowerCase().includes(lowerCasedTerm) || c.display_name.toLowerCase().includes(lowerCasedTerm));
+    });
+};
+
+const filterJoinedChannels = (joinedChannels: MyChannelModel[], allChannels: Channel[] | undefined) => {
+    const ids = joinedChannels.map((c) => c.id);
+    return allChannels?.filter((c) => !ids.includes(c.id));
+};
+
+const makeLeftButton = (icon: ImageResource): OptionsTopBarButton => {
+    return {
+        id: CLOSE_BUTTON_ID,
+        icon,
+        testID: 'close.browse_channels.button',
+    };
+};
+
+const makeRightButton = (theme: Theme, formatMessage: IntlShape['formatMessage']): OptionsTopBarButton => {
+    return {
+        color: theme.sidebarHeaderTextColor,
+        id: CREATE_BUTTON_ID,
+        text: formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        showAsAction: 'always',
+        testID: 'browse_channels.create.button',
+    };
+};
+
+const close = () => {
+    Keyboard.dismiss();
+    dismissModal();
+};
+
+const channelKeyExtractor = (channel: Channel) => {
+    return channel.id;
+};
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        searchBar: {
+            marginHorizontal: 12,
+            borderRadius: 8,
+            marginTop: 12,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+        noResultContainer: {
+            flexGrow: 1,
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center',
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center',
+        },
+        listContainer: {
+            paddingHorizontal: 20,
+            flexGrow: 1,
+        },
+    };
+});
+
+export default function BrowseChannels({
```

Comment:
Hey @larkox any chance this component can be split into multiple smaller components? it has a lot going on and I think it can be simplified a bit, thoughts?

Reply:
I moved the flatlist to a different element, which removed part of the rendering functions, leaving this component just handling the channel search, filter, and selection. There is still the loading render, since we want to show the loading in the center of the screen if there are no elements in place.